pyiceberg/table/__init__.py

```diff
@@ -1978,6 +2013,21 @@ def create_tag(self, snapshot_id: int, tag_name: str, max_ref_age_ms: Optional[i
         self._requirements += requirement
         return self
 
+    def remove_tag(self, tag_name: str) -> ManageSnapshots:
+        """
+        Remove a tag.
+
+        Args:
+            tag_name (str): name of tag to remove
+
+        Returns:
+            This for method chaining
+        """
+        update, requirement = self._transaction._remove_ref_snapshot(ref_name=tag_name)
```

Comment:
add `self._commit_if_ref_updates_exist()` once it's merged into main from #758 .